src/librustc_lint/builtin.rs

```diff
@@ -723,6 +723,45 @@ impl EarlyLintPass for IllegalFloatLiteralPattern {
 }
 
 declare_lint! {
+    pub UNUSED_DOC_COMMENT,
+    Deny,
+    "detects use of useless doc comments"
```

Comment:
I would not say "useless", rather that they will be ignored by rustdoc (theoretically, another tool could use them).